### ISSUE TITLE
Training on custom dataset

### ISSUE THREAD

**Issue by saivineethkumar** [MichiganCOG/TASED-Net#2]
@kylemin 
Hi, I want to train the model on a custom dataset, can you describe the steps involved and details like the structure and format of the training input and annotation data, the training parameters you used for the training etc.

thank you.

**Comment by kylemin** [comment]
Hi,
Thank you for your interest.

1. Training data: Images (224x384) and their matching saliency maps in gray scale (all in PNG format). See more details [here](https://github.com/wenguanwang/DHF1K).
2. Training input: 224x384x32 tensor (32 stacked input images)
3. Training parameters: learning rate (starts from 0.1 and decays twice by a factor of 10 at step 750 and 950), weight decay (2e-7), batch size: 40, momentum of sgd: 0.9

I hope it helps.
Thank you.

**Comment by saivineethkumar** [comment]
@kylemin 
The video data I have is of size 1280x720, can this be used to train the model? 
and also the script to train, is it the one from your other repo "ViP"?

thanks

**Comment by kylemin** [comment]
Our method can work on any size of input data because it is a fully-convolutional network. If there is a memory issue, you can simply resize the input.
Yes, "ViP" is the one. It is easy to use (btw, we plan to release the training codes within the next few updates).

Thank you.

**Comment by saivineethkumar** [comment]
Hi @kylemin 
Can you tell me how did you convert the fixation points into a saliency map for training(as given in the output folder of code )?

thank you.

**Comment by saivineethkumar** [comment]
Hi @kylemin 
Can you please tell how you converted the gaze points into saliency maps for training?

**Comment by kylemin** [comment]
I had exactly the same question, and the answer is in this [discussion](https://github.com/wenguanwang/DHF1K/issues/2).

I hope it helps.

**Comment by saivineethkumar** [comment]
Hi @kylemin,
You said you are going to release the training codes in the next version, can you please tell when can we expect those...

thank you!

**Comment by kylemin** [comment]
Hi,

Sorry for the delay. We will let you know right after training codes are available. ViP is currently integrated with DHF1K dataset, so the remaining job is to implement KLD loss and verify the training really works. I hope it will be soon, within two weeks. Thank you for your patience!

**Comment by prashnani** [comment]
> Dear,
> Thank you for your interests.
> 
>     1. Training data: Images (224x384) and their matching saliency maps in gray scale (all in PNG format). See more details [here](https://github.com/wenguanwang/DHF1K).
> 
>     2. Training input: 224x384x32 tensor (32 stacked input images)
> 
>     3. Training parameters: learning rate (starts from 0.1 and decays twice by a factor of 10 at step 750 and 950), weight decay (2e-7), batch size: 40, momentum of sgd: 0.9
> 
> 
> I hope it helps.
> Thank you.

@kylemin thanks for the clarification!

Could you please clarify if the learning rate for different model components is different? 
From what I read from your paper, the lr is 0.001 for encoder and 0.1 for prediction network? 
If this is the case, could you give a list of which modules in [TASED_v2 class](https://github.com/MichiganCOG/TASED-Net/blob/master/model.py#L4) (in model.py) should have an lr of 0.001?

Also, in your reply, by "step" you mean number of backward passes through the network? 
or is it number of epochs (where one epoch would contain dataset_size/batch_size number of backward passes)?

Thanks!

**Comment by kylemin** [comment]
@saivineethkumar We got your email, so we just released a simple script for training. You can use it to train it on any custom dataset.

**Comment by kylemin** [comment]
@prashnani We just released the training codes, so we believe it can answer your first question.
For the second question, yes, "step" means that the number of backward passes.

Thank you.
